clusterloader2/testing/l4ilb/config-nodepool-upgrade.yaml

```diff
@@ -0,0 +1,124 @@
+# Timer measurement is used to measure the time between the start of nodepool upgrade and when
+# the ILB services are running and reachable.
+# ServiceCreationLatency measurement is used only for determining the moment when all of the
+# ILBs are reachable. Since it reports time based on the initial creation of the service, and
+# not from the nodepool upgrade, the measured time will not be shown.
+
+# Constants
+{{$LB_REPLICAS_PER_NS := DefaultParam .CL2_LB_REPLICAS_PER_NS 5}}
+{{$LB_BACKEND_SIZE := DefaultParam .CL2_LB_BACKEND_SIZE 300}}
+{{$EXTERNAL_TRAFFIC_POLICY := DefaultParam .CL2_EXTERNAL_TRAFFIC_POLICY "Cluster"}}
+{{$ilbWaitTimeout := DefaultParam .CL2_ILB_WAIT_TIMEOUT "10m"}}
+{{$ilbQPS := DefaultParam .CL2_ILB_TEST_QPS 20}}
+{{$NODEPOOL_UPGRADE_LABEL := "ilb-np-upgrade"}}
+{{$namespaces := 1}}
+
+# Command to be executed
+{{$NODEPOOL_UPGRADE_COMMAND := DefaultParam .CL2_EXEC_COMMAND nil}}
+
+name: l4ilb-nodepool-upgrade
+namespace:
+  number: {{$namespaces}}
+tuningSets:
+- name: ILBConstantQPS
+  qpsLoad:
+    qps: {{$ilbQPS}}
+steps:
+- name: Start measurement for running pods
+  measurements:
+  - Identifier: WaitForRunningDeployments
+    Method: WaitForControlledPodsRunning
+    Params:
+      action: start
+      apiVersion: apps/v1
+      kind: Deployment
+      labelSelector: group = ilb-load
+      operationTimeout: 15m
+- name: Creating ILBs
+  phases:
+  - namespaceRange:
+      min: 1
+      max: {{$namespaces}}
+    replicasPerNamespace: {{$LB_REPLICAS_PER_NS}}
+    tuningSet: ILBConstantQPS
+    objectBundle:
+    - basename: ilb-npu-service
+      objectTemplatePath: service.yaml
+      templateFillMap:
+        DeploymentBaseName: ilb-npu-dep
+        ExternalTrafficPolicy: {{$EXTERNAL_TRAFFIC_POLICY}}
+        ILBSizeLabel: {{$NODEPOOL_UPGRADE_LABEL}}
```

Comment:
Why is this label needed? Can't we use "group=ilb-load"?

Reply:
The service.yaml template inside the l4ilb directory uses a size label. We add a value for the size label in order to reuse the template for multiple tests which are using measurements to select services by label.